### DISCUSSION TITLE
Trailing slash problem if request: Rest in parameters

### DISCUSSION THREAD

**Discussion by boumboum** [fastapi/fastapi#8908] [Questions] [answered]
Given that def:
```
@app.get("/allcounters", response_class=HTMLResponse)
async def allcounters(request: Request):
    return templates.TemplateResponse('allcounters.html', context={'request': request, 'cards': counter_data})
```

and the URL looks like that:
`http://localhost:8000/allcounters`

Than it causes this error:
`{"detail":[{"loc":["path","id"],"msg":"value is not a valid integer","type":"type_error.integer"}]}`

But if I change the decorator to that:
`@app.get("/allcounters/", response_class=HTMLResponse)`

and the URL call like that:
`http://localhost:8000/allcounters/`

it works.
I suspect the `request: Request` parameter in my def causing the issue.
The expectation of course would be to get it working without leading slash.
Any clue?

**Comment by Kludex** [comment, marked as answer]
You're probably calling another endpoint? https://fastapi.tiangolo.com/tutorial/path-params/#order-matters

**Comment by boumboum** [comment]
Well... This is a hint. Tomorrow I'll test it in the office

**Comment by boumboum** [comment]
That was the problem. Thanks for the help. The issue can be closed.
